base/util/PKICertImport.bash

```diff
@@ -0,0 +1,333 @@
+#!/bin/bash
+
+# Copyright (C) 2018 Red Hat
+
+# PKICertImport performs a validate-then-import strategy for importing
+# certificates into a NSS DB or HSM Token by creating a copy of the NSS
+# DB prior to import. This ensures that the certificate will not be used
+# until its signature and usage has been verified. It replaces the
+# `certutil -A` and `certutil -V`.
+function PKICertImport() {
+    ## [ overview ] ##
+
+    # This script has four major sections:
+    #
+    #   1. Globals -- the definitions of all script-global variables
+    #   2. Helper functions -- functions which don't perform key operations
+    #   3. Core commands -- functions which interact with the NSS DB via
+    #                       certutil
+    #   4. Program flow -- main flow of the program
+
+
+    ## [ globals ] ##
+
+    # Path to certificate; parsed from arguments.
+    local CERT_PATH=""
+
+    # User-given nickname for the certificate.
+    local CERT_NICKNAME=""
+
+    # Whether or not the certificate is in ASCII/PEM format.
+    local CERT_ASCII="false"
+
+    # What trust flags to use when importing the certificate.
+    local CERT_TRUST=""
+
+    # What usage flags to validate the certificate against.
+    local CERT_USAGE=""
+
+    # Location of the original NSS DB.
+    local NSSDB=""
+
+    # Type of the NSSDB.
+    local NSSDB_TYPE=""
+
+    # Location to the NSS DB Password file, if present.
+    local NSSDB_PASSWORD=""
+
+    # Name of the HSM token, if used.
+    local HSM_TOKEN=""
+
+
+    ## [ helper functions ] ##
+
+    # __e prints error messages, prefixing them with "e: " and writing the
+    # output to stderr instead of stdout.
+    function __e() {
+        echo "e:" "$@" 1>&2
+    }
+
+    # __v prints debug messages in verbose mode; these also go to stderr.
+    # Messages are only present if the environment variable VERBOSE is set.
+    function __v() {
+        if [ "x$VERBOSE" != "x" ]; then
```

Comment:
@ladycfu Do you have a preference about this vs. the `-z` notation? I personally prefer `[ "x$var" != x ]` as I think it is clearer, but `[ ! -z "$var" ]` is a viable option as well. ShellCheck doesn't appear to have a preference.

Reply:
I'm not an expert in this.  In my view, if it works then it's fine. Endi might know better.

Reply:
I don't have any preference. Either way should be fine.